src/components/views/rooms/EventTile.tsx

```diff
@@ -658,6 +659,33 @@ export class UnwrappedEventTile extends React.Component<EventTileProps, IState>
         if (this.context.timelineRenderingType === TimelineRenderingType.Notification) return false;
         if (this.context.timelineRenderingType === TimelineRenderingType.ThreadsList) return false;
 
+        // MSC3952 mentions attempt to avoid notifying the user multiple times
+        // for edits. This is done by not including the same user in the mentions
+        // field for each iteration of the edit (only the first time they're included).
+        //
+        // An edited event should show up as highlights iff the m.new_content includes
+        // a mention to the user, this differs from what push actions occurred on the
+        // event.
+        //
+        // Note that we can't just check if any of the iterations of the event caused
+        // a highlight since that would preclude removing a user mention.
```

Comment:
Thinking on this more, I think this is somewhat bogus logic -- the user could disable the push rule (and it also doesn't handle room mentions properly).

Reply:
The goal of this was to ensure that if an event mentioned a user than it still appear red, even if the latest edit didn't mention the user (which MSC3952 says that users should not continually be included in the mentions field to attempt to avoid duplicate notifications).

The current logic (of handling mentions specially) wouldn't work if:

* The user disables the push rule.
* If there's a room mention (although I'm sure the logic could be extended for that).
* If a *different push rule* would be matched, e.g. for a keyword.

Other ideas of what we could do:

* Use the `m.new_content` instead of `content` when checking push rules? (This sounds not quite with the spec and could break other push rules.)
* Check if any edit to the text would have caused a highlight (essentially take the max notification level across all edits). This will give wrong results if the user was *removed* from the mentioned users. (But the user would still have received a highlight notification, so maybe that doesn't matter?)

Reply:
I discussed this a bit with @daniellekirkwood and she suggested that if *any of the edits* caused a notification then it must be highlighted. This will help with user confusion if they see a red count but cannot find a corresponding message.

We could probably improve this in the future, but it doesn't need to be wrapped up in this change.

Reply:
I wonder if we should just leave this out for now and do this as a separate change since it is somewhat a change in behavior? (And not really related to this PR -- i.e. if a message matches my keyword, then the person edits it and it no longer matches than today we would no longer highlight the message.)

Reply:
Will do a follow-up for this: https://github.com/vector-im/element-web/issues/24377#issuecomment-1474284517